### ISSUE TITLE
Double click handling.

### ISSUE THREAD

**Issue by kajojify** [urwid/urwid#268]
Hello, guys! Thank you for so nice library. You're awesome! :)
Could you please help me?

**My question:** 
Now I can handle mouse events with adding `mouse_event` method to my widget.
But I want to handle double clicking mouse event. 
How can I do it? Is there any way to write something like: 
```
def mouse_event(self, size, event, button, col, row, focus):
    if event == "mouse doubleclick" and button == 1:
       do_something()
```

Kind regards.

**Comment by federicotdn** [comment]
You could try saving the time of the last click to a variable `last_click`, and then when the widget is clicked again, compare the value of `last_click - current_time` to a certain delta value, like 500 milliseconds.  If the value is lower than the delta, then the widget was double-clicked.

**Comment by kajojify** [comment]
@federicotdn That's a nice idea. Thank you very much. So then I don't fully understand, what this PR is about (https://github.com/urwid/urwid/pull/32).

**Comment by and3rson** [comment]
I agree with @federicotdn, it's pretty trivial to implement it manually. Yeah, #32 seems to leverage some curses functionalty for this, not sure why it was not merged. Probably because it only supports actual terminals.
I'll take a look at it tomorrow.

**Comment by and3rson** [comment]
Interesting, it seems to have been merged and I can see it in the upstream, but I cannot reproduce the functionality. I think we can consider this a bug.

**Comment by eliasdorneles** [comment]
In case anyone needs, here is a sample implementation, used for my solitaire game: https://github.com/eliasdorneles/usolitaire/blob/35550af104054bc2e6b030ed385be535f96f8316/usolitaire/ui.py#L95-L104
